pcdswidgets/vacuum/base.py

```diff
@@ -368,33 +516,69 @@ def assemble_layout(self):
             return
         self.clear()
 
-        # (Layout, items)
-        widget_map = {
-            ContentLocation.Hidden: (QVBoxLayout,
-                                     [self.icon]),
-            ContentLocation.Top: (QVBoxLayout,
-                                  [self.controls_frame,
-                                   self.icon]),
-            ContentLocation.Bottom: (QVBoxLayout,
-                                     [self.icon,
-                                      self.controls_frame]),
-            ContentLocation.Left: (QHBoxLayout,
-                                   [self.controls_frame,
-                                    self.icon]),
-            ContentLocation.Right: (QHBoxLayout,
-                                    [self.icon,
-                                     self.controls_frame]),
-        }
-
-        layout = widget_map[self._controls_location][0]()
+        grouped_frame = QFrame(self)
+        grouped_widgets = QVBoxLayout()  # Default
+
+        # Determine what widgets to group
+        if self._text_location in [ContentLocation.Left, ContentLocation.Right] and self._text_location == self._controls_location:
+            grouped_widgets = QVBoxLayout()
+            if self.name:
```

Comment:
Nitpick: various python linters/style guides will suggest you use explicit `is None` or `is not None` checks whenever you expect something to be `None` because objects can evaluate to `False` sometimes (notorious examples including empty lists and blank strings). This isn't one of those cases so this is a nitpick rather than a full suggestion.
```suggestion
            if self.name is not None:
```

Reply:
This makes sense